### ISSUE TITLE
Consertar o dataset.

### ISSUE THREAD

**Issue by Fideliszan** [HaraHeique/TCC-rede-neural-siamesa#18]

**Comment by Fideliszan** [comment]
O dataset tem que conter **72 mil** pares de frases sendo 36 mil frases de mesma autoria (com tag igual a 1) e 36 mil frases de autorias distintas (com tag igual a 0). Vai ficar assim:
1,2, Faulkner1, Faulkner2, 1
............................................................
11999,12000, Faulkner11999, Faulkner12000, 1

12001,12002, Hami12001,Hami12002, 1
..........................................................
23999,24000, Hami23999,Hami24000, 1

24001,24002, Roth24001,Roth24002, 1
..........................................................
35999,36000, Roth35999,Roth36000, 1

36001,36002, Faukner36001,Hami36002, 0
..........................................................
47999,48000, Faukner47999,Hami48000, 0

48001,48002, Faukner48001,Roth48002, 0
..........................................................
59999,60000, Faukner59999,Roth60000, 0

60001,60002, Hami60001,Roth60002, 0
..........................................................
71999,72000, Hami71999,Roth72000, 0

**Comment by HaraHeique** [comment]
Os resultados de acurácia em geral estão dando em média (com 50 epochs):
- **78% no treinamento**
- **66% a 69% na validação** (em torno disto)

## 72000 registros

### 70/30 - 7 e 35 max_seq_length

![history-graph-70-30-50-7-72000](https://user-images.githubusercontent.com/26990166/107160899-d6c21380-6977-11eb-9463-f7975e49371b.png)

![history-graph-70-30-50-35-72000](https://user-images.githubusercontent.com/26990166/107160938-12f57400-6978-11eb-9514-1eec109f8a23.png)

### 90/10 - 7 e 35 max_seq_length

![history-graph-90-10-50-7-72000](https://user-images.githubusercontent.com/26990166/107160944-1d177280-6978-11eb-96fc-cff5439ede24.png)

![history-graph-90-10-50-35-72000](https://user-images.githubusercontent.com/26990166/107160947-20aaf980-6978-11eb-9c95-2feca7317f7c.png)

## 10800 registros

### 70/30 - 7 e 35 max_seq_length

![history-graph-70-30-50-7-10800](https://user-images.githubusercontent.com/26990166/107160958-31f40600-6978-11eb-931b-99c4b92b3f3a.png)

![history-graph-70-30-50-35-10800](https://user-images.githubusercontent.com/26990166/107160965-3ddfc800-6978-11eb-8185-63394f808368.png)

### 90/10 - 7 e 35 max_seq_length

![history-graph-90-10-50-7-10800](https://user-images.githubusercontent.com/26990166/107160973-489a5d00-6978-11eb-9b23-304a48f8047b.png)

![history-graph-90-10-50-35-10800](https://user-images.githubusercontent.com/26990166/107160978-4f28d480-6978-11eb-8eb2-5f8a12c15ed5.png)


O dataset eu criei dois diferentes, onde um foi o combinado de 72000 linhas (36K de similaridade 1 e 36K similaridade 0). Mas também criei outro dataset que possui 10800 linhas (36K de similaridade 1 e 72K de similaridade 0), onde este segundo contem todas os registros de comparação entre os 3 autores, dado que o de 72K linhas não é possível utilizar todos os registros.

Lembrando que todos os testes foram utilizando o loss **mse** e otimizador **Adam** padrão do tensorflow. Conclui-se que realmente melhorou consideravelmente em comparação aos resultados anteriores (60% no treinamento e 52% na validação) entretanto talvez não foi o que esperávamos. Qualquer coisa dê uma olhada no dataset criado:
- [Com 72000 registros](https://github.com/HaraHeique/TCC-rede-neural-siamesa/blob/master/src/data/training/training-72000-sentences.csv);
- [Com 10800 registros](https://github.com/HaraHeique/TCC-rede-neural-siamesa/blob/master/src/data/training/training-sentences.csv).

**Comment by HaraHeique** [comment]
@Fideliszan Link do notebook no colab. Lá contém o log das execuções realizadas.

https://github.com/HaraHeique/TCC-rede-neural-siamesa/blob/master/TCC_rede_neural_siamesa.ipynb
